crates/polars-arrow/src/bitmap/immutable.rs

```diff
@@ -590,3 +591,39 @@ impl From<Bitmap> for arrow_buffer::buffer::NullBuffer {
         unsafe { arrow_buffer::buffer::NullBuffer::new_unchecked(buffer, null_count) }
     }
 }
+
+impl Splitable for Bitmap {
+    #[inline(always)]
+    fn check_bound(&self, offset: usize) -> bool {
+        offset <= self.len()
+    }
+
+    unsafe fn _split_at_unchecked(&self, offset: usize) -> (Self, Self) {
+        let bytes = &self.bytes;
+
+        let ubcc = self.unset_bit_count_cache.load(Ordering::Relaxed);
+
+        let (lhs_ubcc, rhs_ubcc) = if ubcc == 0 {
+            (0, 0)
+        } else if ubcc == self.length as u64 {
+            (offset as u64, (self.length - offset) as u64)
+        } else {
```

Comment:
I want to include this logic from slice:


 ```rust
         if *unset_bit_count_cache >> 63 == 0 {
            // If we keep all but a small portion of the array it is worth
            // doing an eager re-count since we can reuse the old count via the
            // inclusion-exclusion principle.
            let small_portion = (self.length / 5).max(32);
            if length + small_portion >= self.length {
                // Subtract the null count of the chunks we slice off.
                let slice_end = self.offset + offset + length;
                let head_count = count_zeros(&self.bytes, self.offset, offset);
                let tail_count = count_zeros(&self.bytes, slice_end, self.length - length - offset);
                let new_count = *unset_bit_count_cache - head_count as u64 - tail_count as u64;
                *unset_bit_count_cache = new_count;
            } else {
                *unset_bit_count_cache = UNKNOWN_BIT_COUNT;
            }
        }
 ```